src/EditorFeatures/Core/LanguageServer/Handlers/References/FindUsagesLSPContext.cs

```diff
@@ -245,7 +245,7 @@ public override async ValueTask OnReferenceFoundAsync(SourceReferenceItem refere
             }
 
             var declarationFile = await _metadataAsSourceFileService.GetGeneratedFileAsync(
-                _document.Project, symbol, signaturesOnly: true, allowDecompilation: false, cancellationToken).ConfigureAwait(false);
+                _document.Project, symbol, signaturesOnly: true, MetadataAsSourceOptions.Default, cancellationToken).ConfigureAwait(false);
```

Comment:
>MetadataAsSourceOptions [](http://example.com/codeflow?start=65&length=23)

Should this read options from IGlobalOptionService as well?

Reply:
I guess it could just in case, but when `signaturesOnly` is true, then neither option matters

Reply:
Would it make sense to take `MetadataAsSourceOptions?`, remove `signaturesOnly` and interpret the null for MetadataAsSource options as "signaturesOnly"?

Reply:
I think thats probably just as undiscoverable as the current state.

Maybe the best thing is a new API for `GetSignaturesOnlyGeneratedFileAsync` that doesn't take options at all

Reply:
Thought about this more, I think I'll just make these all use `IGlobalOptionService` then we don't need to worry about when new options are added that might be relevant